File: 3_laboratorijska_vjezba/1_1_zadatak/Zadaca/src/main/java/ooup/plugins/Tiger.java
package ooup.plugins;

import ooup.Animal;

public class Tiger extends Animal {
    private String name;

    public Tiger(String name) {
        this.name = name;
    }

    @Override
    public String name() {
        return name;
    }

    @Override
    public String greet() {
        return "grauu";
    }

    @Override
    public String menu() {
        return "meat";
    }
}

File: 3_laboratorijska_vjezba/1_zadatak/tiger.c
#include<stdio.h>
#include<stdlib.h>

typedef const char* (*PTRFUN)();

struct Tiger
{
    PTRFUN* vtable;
    const char* ime;
};

const char* name(void* animal) {
    return ((struct Tiger*)animal)->ime;
}

const char* greet() {
    return "rauuu";
}

const char* menu() {
    return "papagaje";
}

PTRFUN tiger_table[3] = {*name, *greet, *menu};

void* create(const char* ime) {
    struct Tiger* tiger = (struct Tiger*)malloc(sizeof(struct Tiger));

    tiger->ime = ime;
    tiger->vtable = tiger_table;

    return tiger;
}
